web/src/views/FileBrowserView/FileBrowser.vue

```diff
@@ -259,167 +265,175 @@ export default {
-        .map((service) => ({
-          name: service.name,
-          url: `${service.endpoint}${dandiRest.assetDownloadURI(identifier, version, asset_id)}`,
-        }));
-    },
+    function assetMetadataURI(asset_id: string) {
+      return dandiRest.assetMetadataURI(props.identifier, props.version, asset_id);
+    }
 
-    assetMetadataURI(asset_id) {
-      return dandiRest.assetMetadataURI(this.identifier, this.version, asset_id);
-    },
+    function fileSize(item: AssetStats) {
+      return filesize(item.size || 0, { round: 1, base: 10, standard: 'iec' });
+    }
 
-    fileSize(item) {
-      return filesize(item.size, { round: 1, base: 10, standard: 'iec' });
-    },
+    function showDelete(item: AssetStats) {
+      return props.version === 'draft' && !item.folder && (isAdmin.value || isOwner.value);
+    }
 
-    showDelete(item) {
-      return this.version === 'draft' && !item.folder && (this.isAdmin || this.isOwner);
-    },
+    async function getItems() {
+      updating.value = true;
+      const { folders, files, count } = await dandiRest.assetPaths(
+        props.identifier, props.version, location.value, page.value, FILES_PER_PAGE,
+      );
+
+      // Set num pages
+      pages.value = Math.ceil(count / FILES_PER_PAGE);
+
+      // Create items
+      // Parent directory if necessary
+      const newItems = location.value !== rootDirectory
+        ? [{ name: parentDirectory, folder: true }]
+        : [];
+
+      // Add folders
+      newItems.concat(Object.keys(folders).map(
+        (key) => ({ ...folders[key], name: `${key}/`, folder: true }),
+      ).sort(sortByName));
+
+      // Add items
+      newItems.concat(Object.keys(files).map(
+        (key) => {
+          const { asset_id, size } = files[key];
+          const services = getExternalServices(asset_id, key, size || 0);
+          return {
+            ...files[key],
+            name: key,
+            folder: false,
+            services,
+          };
+        },
+      ).sort(sortByName));
+
+      items.value = newItems;
+      updating.value = false;
+    }
 
-    async deleteAsset(item) {
+    async function deleteAsset(item: AssetFile) {
       const { asset_id } = item;
       if (asset_id !== undefined) {
         // Delete the asset on the server.
-        await dandiRest.deleteAsset(this.identifier, this.version, asset_id);
+        await dandiRest.deleteAsset(props.identifier, props.version, asset_id);
 
         // Recompute the items to display in the browser.
-        this.$asyncComputed.items.update();
+        getItems();
       }
-      this.itemToDelete = null;
-    },
+      itemToDelete.value = null;
+    }
 
-    fetchDandiset() {
-      store.dispatch.dandiset.fetchDandiset({
-        identifier: this.identifier,
-        version: this.version,
-      });
-    },
+    // Refresh files/folders whenever URL changes
+    watch(location, getItems);
```

Comment:
This watcher can use `immediate` and the single `getItems` call below can be removed.

```suggestion
    watch(location, getItems, { immediate: true });
```

Reply:
@AlmightyYakob this introduces a bug - running it immediately will trigger another watcher which will result in another call to `getItems` with a blank `path_prefix` (i.e. the file browser will always change to the root folder shortly after the page loads)
![Peek 2022-02-14 14-39](https://user-images.githubusercontent.com/37340715/153942748-89478a72-39c3-4eac-abab-c6ddbb147984.gif)

Notice the file browser initially displays the correct files, but then immediately changes to the files in the root directory. 

It seems this is happening because of a redundant watcher that is causing conflicts - I fixed it in https://github.com/dandi/dandi-archive/pull/882/commits/ad57e24f0cea8be0c011a1e2555c028874d61b3c. LMK if anything is unclear.